### DISCUSSION TITLE
How to output Chinese and correct code

### DISCUSSION THREAD

**Discussion by shijunti19** [rust-shell-script/rust_cmd_lib#27] [Q&A]
![image](https://user-images.githubusercontent.com/5392211/119750582-37943e00-becc-11eb-8d6f-9d8605eead08.png)

run_fun
Chinese is not displayed

![image](https://user-images.githubusercontent.com/5392211/119750698-6d392700-becc-11eb-92a1-b7bb5c66296a.png)

The correct output should be：f:\web\xiaolu\rust\监听命令\a031

I can't transcode with encoding

**Comment by rust-shell-script** [comment]
The Chinese output would be just fine for valid strings, see
https://github.com/rust-shell-script/rust_cmd_lib/blob/0b6e05b9fefe818137fe5044c55e854c7abddadd/tests/test_macros.rs#L94 and
https://github.com/rust-shell-script/rust_cmd_lib/blob/0b6e05b9fefe818137fe5044c55e854c7abddadd/tests/test_macros.rs#L227

The problem is that the output from your script is not valid utf8 strings, however you can `spawn!(1.bat)?.wait_with_pipe(...)` to decode the the raw output for this case. Since I don't have a working windows environment, sorry I can not provide further suggestions.

**Reply by shijunti19** [reply]
@rust-shell-script Does rust have any good transcoding library

How to transfer GBK out of UTF-8

**Reply by rust-shell-script** [reply]
You can try this one: https://stackoverflow.com/questions/45788866/how-to-read-a-gbk-encoded-file-into-a-string (from google)

**Reply by shijunti19** [reply]
@rust-shell-script 
![image](https://user-images.githubusercontent.com/5392211/120129224-2877fd80-c1f6-11eb-9f7a-20e22edf2196.png)
I found it possible

1. The running directory is UTF-8

2. Bat returns GBK

I won't transform
The return should be converted to an object
